src/App.tsx

```diff
@@ -8,53 +8,138 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getUsers } from './api/users';
+import { getPosts } from './api/posts';
+import { Post } from './types/Post';
+import { getComments } from './api/comments';
+import { Comment } from './types/Comment';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [post, setPost] = useState<Post | null>(null);
+  const [loadingPosts, setLoadingPosts] = useState(false);
+  const [errorLoadPosts, setErrorLoadPosts] = useState<boolean>(false);
+  const [openPostDetails, setOpenPostDetails] = useState<number>(0);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [loadingComments, setLoadingComments] = useState(false);
+  const [errorLoadComments, setErrorLoadComments] = useState<boolean>(false);
+  const [writeComment, setWriteComment] = useState<boolean>(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
 
-              <Loader />
+  function getPostsUser(user: User) {
+    setLoadingPosts(true);
+    getPosts(user.id)
+      .then(setPosts)
+      .catch(() => {
+        setErrorLoadPosts(true);
+      })
```

Comment:
Consider adding more detailed error messages or logging within the `.catch()` block to aid in debugging and provide more context about the error.